### PR TITLE
exe_path() and related return Result with a specific error so downstr…

### PR DESCRIPTION
**Pull request by RCasatta**
…eam is easier to understand the error

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -68,12 +68,32 @@ pub enum P2P {
...
 pub enum Error {
     /// Wrapper of io Error
     Io(std::io::Error),
     /// Wrapper of bitcoincore_rpc Error
     Rpc(bitcoincore_rpc::Error),
+    /// Returned when calling methods requiring a feature to be activated, but it's not
+    NoFeature,
+    /// Returned when calling methods requiring a env var to exist, but it's not
+    NoEnvVar,
+    /// Returned when calling methods requiring a feature or anv var, but both are absent
```
Comment (LeoComandini) [inline src/lib.rs:80]:
```suggestion
    /// Returned when calling methods requiring either a feature or env var, but both are absent
```

File: src/lib.rs [hunk 2]
```diff
@@ -68,12 +68,32 @@ pub enum P2P {
...
+    NeitherFeatureNorEnvVar,
+    /// Returned when calling methods requiring a feature or anv var, but both are absent
```
Comment (LeoComandini) [inline src/lib.rs:82]:
```suggestion
    /// Returned when calling methods requiring either a feature or env var, but both are present
```

**Comment by LeoComandini** [comment]
utACK efed1ace0cfe8bd433308e1786c3dec23e33ea58

File: src/lib.rs (hunk 2 above)
Comment (RCasatta) [reply src/lib.rs]: ta

File: src/lib.rs (hunk 1 above)
Comment (RCasatta) [reply src/lib.rs]: ta

**Pull request merged**